Userland/Libraries/LibC/cxxabi.cpp

```diff
@@ -59,15 +59,15 @@ static bool atexit_region_should_lock = false;
 
 static void lock_atexit_handlers()
 {
-    if (atexit_region_should_lock && mprotect(atexit_entries, atexit_region_bytes(), PROT_READ) < 0) {
+    if (atexit_entries != nullptr && atexit_region_should_lock && mprotect(atexit_entries, atexit_region_bytes(), PROT_READ) < 0) {
```

Comment:
This is incredibly suspicious.

I can see a few possible explainations:
1) we forgot to initialize bss as a zero page, and atexit_entries was never set to a null value for the first exit time destructor call
2) someone external to this file tried to lock the handlers before anyone called a global initializer
3) we never call global initializers?

Does crt0.cpp (non shared) walk the initializers properly?

Reply:
> Does crt0.cpp (non shared) walk the initializers properly?

It doesn't appear like it, we just mark `s_global_initializers_ran = true` and go on with with our life. I guess that's a side effect of never actually testing crt0.cpp outside of the DynamicLoader.